preprocess.py

```diff
@@ -90,49 +48,45 @@ def dict_path(lang):
         return dest_path("dict", lang) + ".txt"
 
     if args.joined_dictionary:
-        assert not args.srcdict, "cannot combine --srcdict and --joined-dictionary"
-        assert not args.tgtdict, "cannot combine --tgtdict and --joined-dictionary"
-        src_dict = build_dictionary(
-            {train_path(lang) for lang in [args.source_lang, args.target_lang]},
-            args.workers,
-        )
+        if args.srcdict:
+            src_dict = task.load_dictionary(args.srcdict)
+        elif args.tgtdict:
+            src_dict = task.load_dictionary(args.tgtdict)
```

Comment:
Cool, this is a nice feature, but let's also make sure the user doesn't specify both srcdict and tgtdict in this case:
```
assert not args.srcdict or not args.tgtdict, "cannot use both --srcdict and --tgtdict with --joined-dictionary"
```

Reply:
Done https://github.com/pytorch/fairseq/pull/448/commits/037aae29e44e2bba41ef3ee5d858e8575aa6c249